src/robusta/core/discovery/discovery.py

```diff
@@ -254,6 +256,39 @@ def discovery_process() -> DiscoveryResults:
             )
             raise e
 
+        # discover helm state
+        helm_releases_map: dict[str, HelmRelease] = {}
+        try:
+            continue_ref: Optional[str] = None
+            for _ in range(DISCOVERY_MAX_BATCHES):
+                secrets = client.CoreV1Api().list_secret_for_all_namespaces(label_selector=f"owner=helm",
+                                                                            _continue=continue_ref)
+                if not secrets.items:
+                    logging.debug(
+                        "[Discovery] No helm data available",
+                        exc_info=True,
+                    )
+                    break
+
+                for secret_item in secrets.items:
+                    if not secret_item.data or not secret_item.data.get("release", None):
+                        continue
+
+                    decoded_release_row = HelmRelease.from_api_server(secret_item.data['release'])
```

Comment:
I'm a little worried about this code.
Id has a lot of decoding and similar.
If one helm release has corrupted or different encoded data, discovery will fail completely

I suggest wrapping the risky line with try/except, so that if one is corrupted, only this will be missing.
```
try"
  decoded_release_row = HelmRelease.from_api_server(secret_item.data['release'])
  # we use map here to deduplicate and pick only the latest release data
  helm_releases_map[decoded_release_row.get_service_key()] = decoded_release_row
except Exception:
  logging.error(....)
```